libvast/src/system/explore_command.cpp

```diff
@@ -0,0 +1,160 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#include "vast/system/explore_command.hpp"
+
+#include "vast/concept/parseable/to.hpp"
+#include "vast/concept/parseable/vast/expression.hpp"
+#include "vast/defaults.hpp"
+#include "vast/detail/make_io_stream.hpp"
+#include "vast/error.hpp"
+#include "vast/format/json.hpp"
+#include "vast/logger.hpp"
+#include "vast/scope_linked.hpp"
+#include "vast/system/atoms.hpp"
+#include "vast/system/explorer.hpp"
+#include "vast/system/node_control.hpp"
+#include "vast/system/read_query.hpp"
+#include "vast/system/signal_monitor.hpp"
+#include "vast/system/sink.hpp"
+#include "vast/system/spawn_or_connect_to_node.hpp"
+#include "vast/system/start_command.hpp"
+#include "vast/system/tracker.hpp"
+
+#include <caf/actor.hpp>
+#include <caf/event_based_actor.hpp>
+#include <caf/scoped_actor.hpp>
+#include <caf/settings.hpp>
+#include <caf/stateful_actor.hpp>
+
+#include <chrono>
+#include <csignal>
+
+using namespace caf;
+using namespace std::chrono_literals;
+
+namespace vast::system {
+
+caf::message
+explore_command(const command::invocation& invocation, caf::actor_system& sys) {
+  VAST_DEBUG_ANON(invocation);
+  const auto& options = invocation.options;
+  // Get query.
+  if (invocation.arguments.size() != 1) {
+    return caf::make_message(make_error(ec::invalid_query,
+                                        "expected query as first and only "
+                                        "positional argument"));
+  }
+  auto query = invocation.arguments.at(0);
+  auto before = caf::get_or(options, "explore.before", vast::duration{0s});
+  auto after = caf::get_or(options, "explore.after", vast::duration{0s});
+  // TODO: Add some way for users to pass "infinity" for either value.
```

Comment:
Side note: Should be easy to add this as a language feature in the timespan parser and have it return `caf::infinite`.

Reply:
Sadly, I think the timespan parser is provided by CAF so its not trivial to change. But maybe it would make sense to change that anyways, due to ch16776.

Reply:
We could accept strings here and use `to<vast::duration>` from parseable to parse the time.

Reply:
Good idea - I've added it to the linked story, so we can do it for both the eraser and the explorer uniformly.